src/lib/i18next.js

```diff
@@ -0,0 +1,60 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2021 Graz University of Technology.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+// I18next dependencies
+// npm install react-i18next i18next i18next-browser-languagedetector --save
+
+import i18next from 'i18next';
+
+// import translations js
+// import { TRANSLATIONS_DE } from './translations/de/translations';
+// import { TRANSLATIONS_EN } from './translations/en/translations';
+
+// import translations json
+// (Note that you need @rollup/plugin-json to import JSON files)
+// rollup.config.js
+//  - import json from "@rollup/plugin-json";
+//  - json(),
+import TRANSLATIONS_DE from './translations/de/translations.json';
+import TRANSLATIONS_EN from './translations/en/translations.json';
+
+
+// https://github.com/i18next/i18next-browser-languageDetector
+import LanguageDetector from 'i18next-browser-languagedetector';
+import { initReactI18next } from 'react-i18next';
+
+i18next
+  // detect user language
+  // learn more: https://github.com/i18next/i18next-browser-languageDetector
+  .use(LanguageDetector)
+  // pass the i18n instance to react-i18next.
+  .use(initReactI18next)
+  // init i18next
+  // for all options read: https://www.i18next.com/overview/configuration-options
+  .init({
+    fallbackLng: 'en',
+    debug: true,
+    resources: {
+      en: {
+        translation: TRANSLATIONS_EN,
+      },
+      de: {
+        translation: TRANSLATIONS_DE,
+      },
+    },
+    // specify language detection order
+    detection: {
+      order: ['htmlTag'],
```

Comment:
How does this detection work?  From Jinja we embed the selected user language? What happens if you change the language on the deposit form itself? We reload the page?

Reply:
This order detects the change in the `htmlTag (add html language tag <html lang="LANGUAGE" ...)`, there are several orders available I can investigate for our need. https://github.com/i18next/i18next-browser-languageDetector#introduction

yes in this detection order - if you change the language on the deposit form it will reload the page.